workbench-core/repo-scripts/github-oidc/src/github-oidc-stack.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CfnOutput, Stack, StackProps } from 'aws-cdk-lib';
+import { FederatedPrincipal, ManagedPolicy, OpenIdConnectProvider, Role } from 'aws-cdk-lib/aws-iam';
+import { NagSuppressions } from 'cdk-nag';
+import { Construct } from 'constructs';
+
+export interface GitHubOIDCStackProps extends StackProps {
+  gitHubOrg: string;
+  gitHubRepos: string[];
+  idp: OpenIdConnectProvider;
+}
+
+export class GitHubOIDCStack extends Stack {
+  public constructor(scope: Construct, id: string, props: GitHubOIDCStackProps) {
+    super(scope, id, props);
+
+    //Create GitHubOIDC Role
+    props.gitHubRepos.forEach((gitHubRepo) => {
+      const githubOIDCRole: Role = new Role(this, `${props.gitHubOrg}-${gitHubRepo}-GitHub-OIDC-Role`, {
+        assumedBy: new FederatedPrincipal(
+          props.idp.openIdConnectProviderArn,
+          {
+            StringEquals: {
+              'token.actions.githubusercontent.com:aud': 'sts.amazonaws.com'
+            },
+            StringLike: {
+              'token.actions.githubusercontent.com:sub': `repo:${props.gitHubOrg}/${gitHubRepo}:*`
+            }
+          },
+          'sts:AssumeRoleWithWebIdentity'
+        ),
+        managedPolicies: [ManagedPolicy.fromAwsManagedPolicyName('AdministratorAccess')]
```

Comment:
Do we need Admin access for deployments?

Reply:
Hmm could we explore this? https://stackoverflow.com/questions/57118082/what-iam-permissions-are-needed-to-use-cdk-deploy

Looks like with CDK we can hand it off to cdk-bootstrap

Reply:
I did explore this, we need permissions for:
1. cdk boostrap
2. cdk deploy
3. and then permission for each resource being deployed.

this can definitely be achieved, it's just time consuming.

The only question I have is, if we are using this role just for internal deployment and integration test then is it ok to use `Admin Access` ? 

If the answer is `no` then I will allocate sometime to setup a role with least privilege.